src/pages/Book.jsx

```diff
@@ -1,15 +1,99 @@
-import { Link, useLocation } from "react-router-dom";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
 import "../styles/Book.css";
+import { useEffect, useState } from "react";
 
 function Book() {
+  const { id } = useParams();
   const location = useLocation();
   const { book } = location.state;
+  const [chapters, setChapters] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const getListOfChapters = async () => {
+      try {
+        const res = await axios.post(
+          "https://api-preprod.lelivrescolaire.fr/graph",
+          {
+            query:
+              "query chapters($bookId:Int){viewer{chapters(bookIds:[$bookId]){hits{id title url valid}}}}",
+            variables: {
+              bookId: parseInt(id),
+            },
+          },
+          {
+            headers: {
+              "Content-Type": "application/json; charset=utf-8",
+            },
+          }
+        );
+        return res.data;
+      } catch (err) {
+        console.error("Error:", err.response ? err.response.data : err.message);
+        return null;
+      }
+    };
+
+    const fetchData = async () => {
+      const chapterslist = await getListOfChapters();
```

Comment:
If there is an error in `getListOfChapters`, the function will correctly catch it and return null, but that will trigger another error on the next line rather than execute the `else` block.

I think Alex's suggestion was for `getListOfChapters` to be a static function (outside the component).

Reply:
I see you did just that for `getListOfBooks`. You could do the same here.